### ISSUE TITLE
Improve docstrings for Blocks

### ISSUE THREAD

**Issue by KronosTheLate** [MakieOrg/Makie.jl#2928]
Blocks generally seem to following the same style for docstrings. For Axis, we have:
```
julia> Base.Docs.getdoc(subtypes(Makie.Block)[1])
  Axis <: Block
  ≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡

  A 2D axis which can be plotted into.

  Constructors
  ==============

  Axis(fig_or_scene; palette = nothing, kwargs...)


  Examples
  ==========

  ax = Axis(fig[1, 1])


  Attributes
  ============

  Axis attributes:

    •      # list of attributes

  Layout attributes:

    •      # list of attributes
```

This is pretty good. But I think there are several points of improvement:
- The first lines, `Axis <: Block` with underline, are noisy and not helpful.
- The example used a figure it does not define, meaning that it does not run.
- It is a quite verbose, and using it in the REPL involves a lot of scrolling. This is mainly because of the explanatory sentence with each attribute. To be consistent with the docstring for plotting functions, I feel it would be better to use autogenerated lists of attributes, with default values (see https://github.com/MakieOrg/Makie.jl/issues/2927). The fuller description can then be found in the online docs.
- Descriptive sentence, examples, and constructors are missing for a lot of Blocks, such as `Slider` and `Box`.

 I feel like the docstring could be slimmed down and improved to something like
```
help?> Slider
search: Slider SliderGrid labelslider! labelslidergrid! IntervalSlider splitdrive
    Axis(fig_or_scene)                              # Just start with the constructors, as for functions showing methods.
    Axis(fig_or_scene, kwargs...)              # show multiple constructors, starting with simple ones. This is done for some docstrings already.

A 2D axis which can be plotted into.

#  Examples
julia> fig = Figure()

julia> ax = Axis(fig[1, 1]);  # Making an axis inside `fig`

julia> xs = 1:10;

julia> my_line = lines!(ax, xs, xs.^2);  # Plotting into the axis

# Positional arguments:
    - `fig_or_scene`: where the axis is placed

# Keyword argument:
    # Autogenerated list of attributes
```

**Suggestion: Go through the docstring for all Blocks, adding descriptions, contructors and examples. Aslo, add a style of argument/sttribute listing that is much less verbose, and consistent with plotting functions.**

**Comment by asinghvi17** [comment]
Agreed, this sounds nice.  Perhaps the general `block_docs` function should be removed and replaced by some docstring extensions for e.g. attribute docs, so we can have a lot more customizability.

I think #2894 might also be a good step towards this.

**Comment by KronosTheLate** [comment]
I am not so fluent on the internals/structures/functions at play here. What is meant by "docstring extensions", how does it add custumizability, and what custumizability is needed? It is something like a function that is given some block, say `Slider`, and returns a list of attributes and default values? And the custumizability is something like the ability to select attributes that are specific to `Slider`, or ones that are generic for `Block`s? That sounds smart, and while methods could be defined for `Block`s, I am not sure if such a framework fits in at all with documenting the plotting functions.

**Comment by KronosTheLate** [comment]
See my comment on reference docstrings in https://github.com/MakieOrg/Makie.jl/issues/2927#issuecomment-1537185320, as it is just as relevant for this issue.

**Comment by jkrumbiegel** [comment]
Improvements to Block docs were just merged two days ago, maybe compare with master first :)

**Comment by KronosTheLate** [comment]
Nice. Is there some way to easily show what the target is, something like suggested in https://github.com/MakieOrg/Makie.jl/issues/2927#issuecomment-1537185320, that would make it easier for others to help with the documentation task? Currently I am not clear on what the target is, and what is missing, and I do not know how to get an overview other than a) doing a real deep dive into master, that would take some time, or b) asking you guys, but you have more important things to do.

**Comment by jkrumbiegel** [comment]
> what the target is

Do you mean in what direction I want to go? This PR was the first step https://github.com/MakieOrg/Makie.jl/pull/2894/files

and I've added a lot more examples to axis.jl, while at the same time deleting corresponding information out of axis.md. The idea being that with attributes being the central thing in Makie, it should be easy to find out for each attribute what to do with it. If you want to help, you can add examples like I did in `axis.jl` to any other `Block` (not plotting functions currently). All the available attributes and their own docstrings are in `types.jl`, where I've also made some improvements, explained more about the possible input types etc.

**Comment by KronosTheLate** [comment]
Thanks for the input. I make no promises, but I will try to find the time to look into your changes and what is missing.
